mlflow/pyfunc/__init__.py

```diff
@@ -1260,6 +1257,22 @@ def spark_udf(
         is_spark_in_local_mode or should_use_nfs or is_spark_connect
     )
 
+    # For spark connect mode,
+    # If client code is executed in databricks runtime and NFS is available,
+    # In driver side, we save model to NFS temp directory,
+    # then in executor side we can load model from the NFS temp directory.
+    should_spark_connect_use_nfs = is_in_databricks_runtime() and should_use_nfs
+
+    if (
+        is_spark_connect
+        and env_manager in (_EnvManager.VIRTUALENV, _EnvManager.CONDA)
+        and not should_spark_connect_use_nfs
+    ):
+        raise MlflowException.invalid_parameter_value(
+            f"In non-databricks runtime or NFS is not available, "
```

Comment:
what is non-databricks runtime?

Reply:
e.g., user runs spark connect client code on their own laptop to connect to databricks shared cluster.

Reply:
is the error message good ? then I will update test to match new error message